File: leetcode/HashTable/leetcode-0349/349.go
package hashtable

func intersection(nums1 []int, nums2 []int) []int {
	res := []int{}

	m := make(map[int]int)

	for _, v := range nums1 {
		m[v] = 1
	}

	for _, v := range nums2 {
		if m[v] > 0 {
			res = append(res, v)
			m[v] = 0
		}
	}

	return res
}

File: leetcode/HashTable/leetcode-0349/349.java
import java.util.HashSet;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        List<Integer> res = new ArrayList<>();
        HashSet<Integer> set = new HashSet<>();

        for (int num : nums1) {
            if (!set.contains(num)) {
                set.add(num);
            }
        }

        for (int num : nums2) {
            if (set.contains(num)) {
                res.add(num);
                set.remove(num);
            }
        }

        int[] ret = new int[res.size()];
        for (int i = 0; i < res.size(); i++) {
            ret[i] = res.get(i);
        }
        return ret;
    }
}